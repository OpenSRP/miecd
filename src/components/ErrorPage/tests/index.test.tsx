import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import { ErrorPage } from '..';

describe('ErrorPage', () => {
    // eslint-disable-next-line jest/expect-expect
    it('renders without crashing', () => {
        mount(<ErrorPage />);
    });

    it('renders correctly', () => {
        const errorTitle = 'Error Title';
        const errorMessage = 'Error Message';
        const props = {
            title: errorTitle,
            message: errorMessage,
        };
        const wrapper = mount(<ErrorPage {...props} />);

        expect(toJson(wrapper)).toMatchSnapshot('full snapshot');

        expect(wrapper.text()).toMatchInlineSnapshot(`"An error occurredError TitleError Message"`);

        expect(wrapper.text().includes(errorTitle)).toBeTruthy();
        expect(wrapper.text().includes(errorMessage)).toBeTruthy();
    });
});
