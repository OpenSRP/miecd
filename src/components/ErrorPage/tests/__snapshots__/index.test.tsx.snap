// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPage renders correctly: full snapshot 1`] = `
<ErrorPage
  message="Error Message"
  title="Error Title"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-md-12"
      >
        <div
          className="error-template"
        >
          <h1>
            An error occurred
          </h1>
          <h2>
            Error Title
          </h2>
          <div
            className="error-details"
          >
            Error Message
          </div>
        </div>
      </div>
    </div>
  </div>
</ErrorPage>
`;
